src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs

```diff
@@ -190,6 +182,29 @@ internal void AppendMetadata(MethodBodyStreamEncoder methodBodyEncoder)
             }
         }
 
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2072:DynamicallyAccessedMembers", Justification = "Members are retrieved from internal cache")]
+        private void WriteInterfaceImplementations(TypeBuilderImpl typeBuilder, TypeDefinitionHandle typeHandle)
+        {
+            if (typeBuilder._interfaces != null)
+            {
+                foreach (Type iface in typeBuilder._interfaces)
+                {
+                    _metadataBuilder.AddInterfaceImplementation(typeHandle, GetTypeHandle(iface));
+                }
+            }
+
+            if (typeBuilder._interfaceMappings != null)
+            {
+                foreach (InterfaceMap map in typeBuilder._interfaceMappings.Values)
+                {
+                    for (int i = 0; i < map._interfaceMethods.Count; i++)
+                    {
+                        _metadataBuilder.AddMethodImplementation(typeHandle, GetMemberHandle(map._targetMethods[i]), GetMemberHandle(map._interfaceMethods[i]));
```

Comment:
Is there anything we can assert to verify the `i`th index is pointing at the correct data across the two collections?

Reply:
The elements in the collections are validated (that their return type and parameters are same etc) before added to the collections simultaneously, and the collections are lists, so the added orders will be kept. If we want to verify here that `i` th elements are belong we would need to do the similar validation again, let me know if you think that is necessary.

Reply:
Adding the Assert and comment, maybe I should use list of value tuples instead of 2 separate lists and match indexes, what do you prefer?

Reply:
Updated to use List<ValueTuple<MethodInfo, MethodInfo>> instead